### PR TITLE
Escape built-in JS proper names

### PR DESCRIPTION
**Pull request by garyb**
Resolves purescript/purescript-random#4, purescript/purescript-random#5

### CODE REVIEW

**Comment by jutaro** [comment]
Just stumbled about this problem when using my own random patch. I'm sorry I've not tested it properly, AND I'm happy the problem is resolved more fundamentally, as it is an evil side effect.

File: src/Language/PureScript/CodeGen/JS/Common.hs [hunk 1]
```diff
@@ -135,7 +135,47 @@ nameIsJsReserved name =
...
-moduleNameToJs (ModuleName pns) = intercalate "_" (runProperName `map` pns)
+moduleNameToJs (ModuleName pns) =
+  let name = intercalate "_" (runProperName `map` pns)
+  in if properNameIsJsReserved name then "$$" ++ name else name
+
+-- |
+-- Checks whether a proper name is reserved in Javascript.
+--
+properNameIsJsReserved :: String -> Bool
+properNameIsJsReserved name =
```
Comment (michaelficarra) [inline src/Language/PureScript/CodeGen/JS/Common.hs]: "Reserved" already has a meaning in JavaScript. How about `isJsBuiltin`? Also, do we want to catch name conflicts with built-ins such as `parseInt` and `setTimeout`?

Comment (garyb) [inline src/Language/PureScript/CodeGen/JS/Common.hs]: Both good points, I'll update the list to contain those as well, but I'll need to fix #824 first.

**Comment by paf31** [comment]
:+1: Looks good. :ship:

**Comment by garyb** [comment]
Will merge this now then to fix the immediate problem in `random`, and then revise based on @michaelficarra's suggestions when I look at #824.

**Pull request merged**